ui/StatusQ/src/leftjoinmodel.cpp

```diff
@@ -0,0 +1,219 @@
+#include "StatusQ/leftjoinmodel.h"
+
+#include <QDebug>
+
+#include <algorithm>
+
+LeftJoinModel::LeftJoinModel(QObject* parent)
+    : QIdentityProxyModel{parent}
+{
+}
+
+void LeftJoinModel::initializeIfReady()
+{
+    if (m_leftModel && m_rightModel && !m_joinRole.isEmpty()
```

Comment:
The models could be deleted at any time. I think we need to connect to the destroyed event and clean the pointers and also check the pointers for null (at leaset in `data`).

Reply:
That's very good point! Thanks for spotting that. I will cover it and add tests.